docs/Roadmap.md

```diff
@@ -0,0 +1,112 @@
+
+1. **Probe and Notify**. Aims to provide the ability to easily **probe** servers, services and API endpoints reliably and **notify** on status updates.
+
+2. **Open & Extensible**. It aims to be **Open** and follow Open standards that allow it to be integrated into **extensible-development** platforms (such as EaseMesh).
+
+3. **Cloud Native**. It is designed to be **cloud-native** compliant. It is scalable, resilient, manageable, and observable and it's easy to integrate with cloud-native architectures of any type.
+
+4. **Predictability & Reliability**. EaseProbe is designed to follow best practices allowing predictable and reliable operations. Even when EaseProbe fails, it aims to fail in a predictable way which allows for easy and speedy troubleshooting of problems.
+
+
+## Features
+The project principles, of EaseProbe' features are separated into two main categories: probe-specific & notify-specific.
+
+### Probe specific
+* HTTP for testing connectivity and validity of responses
+* TCP for testing connectivity
+* Shell for executing custom probe scripts
+* SSH for remote ssh commands
+* Host for CPU, Memory, Disk usage metrics
+* Clients for Redis, MySQL, MongoDB, PostgreSQL, Kafka, Zookeeper
+
+### Notify specific
+* Mail notification
+* AWS SNS notification
+* Log files 
+* Slack
+* Discord
+* Telegram
+* WeChat Work
+* Lark
+* DingTalk
+* Twilio
+* Nexmo
+* YunPian
+
+... with new notification backends been added constantly.
+
+## Roadmap 2022
+Some of the features that are planned/considered for 2022 can be broken down into three categories: *General*, *Probe*, *Notify*
+
+
+### General
+* Work on detailed documentation
+* support for common daemon features
+  * [x] megaease/easeprobe#75 add `daemon()` & `/var/run/easeprobe.pid` support
+  * [x] megaease/easeprobe#75 add SIGHUP, and ensure it closes and re-opens of logfile to allow for `easeprobe.log` rotation
+  * add syslog support eg `logfile: syslog`
+* Support for common `timeformat`, use standard timezone and `strftime` conversions, eg `timezone: [UTC|local|Europe/Athens]`, `timeformat: %F %R:%S UTC`
+* Add opt-out options where appropriate
+  * [x] megaease/easeprobe#75 add opt-out option for `logfile` option 
+  * Add opt-out option for SLA data persistence `data: false`
+  * Make historical data configurable `history: false` and avoid creating backups of statistics
+
+### Probes
+* integrate automatic service discovery which includes probing details
+* add plain old icmp ping probe (or read next)
+* shell probe command must allow allocation or not of `stdout`/`stderr` or redirection of them (eg right now we are forced to create wrapper scripts to achieve this) This will also make ping by shell usable
+* Add support to define desired notification on probes???
+```yaml
+tcp:
+  - name: Memcached
+    host: 10.7.0.253:11211
+    notify: "Server #Alert" # notify.discord.name
+```
+* Add support for host group probes (eg 1 host definition with 4 services)
```

Comment:
Terribly sorry for leaving these details out, i was having a hard time to describe it. 

So the idea is that a service can have more than one systems serving it such as a cluster of load balanced webservers. The actual SLA and certain reports should not address the individual "systems" but rather the overall performance of the web service across the servers. 

Alternative: 1 server with 4 different services, all working under the name of `MyServer`, when i get reports about problems i still get to see the individual failures but my daily reports will now have these 4 different services under a single grouping.

I hope this clears up what i had in mind here.

Reply:
it's quite clear. And I could imagine some statistical analysis works, and it could have other similar scenarios.  So, I think we need a general abstraction solution here like a math formula. Keep thinking hard ...